libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/BotFrameworkHttpClient.cs

```diff
@@ -42,51 +54,61 @@ public class BotFrameworkHttpClient : BotFrameworkClient
             IChannelProvider channelProvider = null,
             ILogger logger = null)
         {
-            HttpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
-            CredentialProvider = credentialProvider ?? throw new ArgumentNullException(nameof(credentialProvider));
-            ChannelProvider = channelProvider;
-            Logger = logger ?? NullLogger.Instance;
-            ConnectorClient.AddDefaultRequestHeaders(HttpClient);
+            if (credentialProvider == null)
+            {
+                throw new ArgumentNullException(nameof(credentialProvider));
+            }
+
+            _httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
+            _logger = logger ?? NullLogger.Instance;
+
+            ConnectorClient.AddDefaultRequestHeaders(_httpClient);
+
+            OriginatingAudience = channelProvider != null && channelProvider.IsGovernment() ? GovernmentAuthenticationConstants.ToChannelFromBotOAuthScope : AuthenticationConstants.ToChannelFromBotOAuthScope;
+
+            _buildCredentialsAsyncImpl = async (appId, oAuthScope) =>
+            {
+                var appPassword = await credentialProvider.GetAppPasswordAsync(appId).ConfigureAwait(false);
```

Comment:
strong assumption: credentials are retrieved using passwords. Unfortunately we ahve other ways to get credentials (certs, MSI, etc)

Reply:
Password auth code has been removed, and took a dependency on the PR https://github.com/microsoft/botbuilder-dotnet/pull/5308 to do authentication using a client that is created by calling `BotFrameworkAuthentication.CreateBotFrameworkClient`.